src/software/ai/navigator/obstacle/new_obstacle/circle_obstacle.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "software/ai/navigator/obstacle/new_obstacle/obstacle.h"
+#include "software/new_geom/circle.h"
+
+/**
+ * An Obstacle represented by a Circle
+ */
+class CircleObstacle : public Obstacle
+{
+   public:
+    CircleObstacle() = delete;
+
+    /**
+     * Construct a circle obstacle by with a Circle
+     *
+     * @param circle Circle to make obstacle with
+     */
+    explicit CircleObstacle(const Circle circle);
+
+    bool contains(const Point& p) const override;
+    double minDistance(const Point& p) const override;
+    bool intersects(const Segment& segment) const override;
+    std::string toString(void) const override;
```

Comment:
I'm not sure what conventions your codebase follows, but having the `void` parameter is equivalent to not having it there.

Reply:
our codebase is a mix of C and C++ and the `void` argument does make a difference for C, so for simplicity we prefer having the `void` argument everywhere

Reply:
I see, that makes sense.